src/jit/scopeinfo.cpp

```diff
@@ -154,6 +154,81 @@ CodeGenInterface::siVarLoc::siVarLoc(const LclVarDsc* varDsc, regNumber baseReg,
     }
 }
 
+//------------------------------------------------------------------------
+// Equals: Compares first reference and then values of the structures.
+//
+// Arguments:
+//    lhs   - a "siVarLoc *" to compare.
+//    rhs   - a "siVarLoc *" to compare.
+//
+// Notes:
+//    Return true if both are nullptr.
+//
+// static
+bool CodeGenInterface::siVarLoc::Equals(const siVarLoc* lhs, const siVarLoc* rhs)
+{
+    if (lhs == rhs)
+    {
+        // Are both nullptr or the same reference
+        return true;
+    }
+    if ((nullptr == lhs) || (nullptr == rhs))
+    {
+        // Just one of them is a nullptr
+        return false;
+    }
+
+    if (lhs->vlType == rhs->vlType)
+    {
```

Comment:
```
if (lhs->vlType != rhs->vlType)
{
    return false;
}
assert(lhs->vlType == rhs->vlType);
switch (lhs->vlType)
{

}
```

```

Reply:
What is the reason of this kind of asserts? Avoid mistakes on future changes?

Reply:
Done!

Reply:
> What is the reason of this kind of asserts? Avoid mistakes on future changes?

Yes. It is a form of comment.  It is not a strict requirement and we do not have many places in jit where we comapre two values using switch, but I thinks it reads better.